staging/src/k8s.io/apimachinery/pkg/util/net/http.go

```diff
@@ -206,52 +206,67 @@ func GetHTTPClient(req *http.Request) string {
 	return "unknown"
 }
 
-// SourceIPs splits the comma separated X-Forwarded-For header or returns the X-Real-Ip header or req.RemoteAddr,
-// in that order, ignoring invalid IPs. It returns nil if all of these are empty or invalid.
+// SourceIPs splits the comma separated X-Forwarded-For header and joins it with
+// the X-Real-Ip header and/or req.RemoteAddr, ignoring invalid IPs.
+// The X-Real-Ip is omitted if it's already present in the X-Forwarded-For chain.
+// The req.RemoteAddr is always the last IP in the returned list.
+// It returns nil if all of these are empty or invalid.
 func SourceIPs(req *http.Request) []net.IP {
```

Comment:
It looks like the only other usage of this is for trace logging, e.g. here:

https://github.com/kubernetes/kubernetes/blob/b008eda8b2dc0fdd884cb56065f7f1667a970a16/staging/src/k8s.io/apiserver/pkg/endpoints/handlers/get.go#L52

Did you do a survey?

Reply:
I did a search, is that what you mean? The ClientIP logic doesn't change.

Reply:
Yup, thanks.